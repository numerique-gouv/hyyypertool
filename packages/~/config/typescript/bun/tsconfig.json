{
  "$schema": "https://json.schemastore.org/tsconfig",
  "display": "Bun",
  "extends": "../base/tsconfig.json",
  "compilerOptions": {
    "types": ["bun-types"]
  }
}
