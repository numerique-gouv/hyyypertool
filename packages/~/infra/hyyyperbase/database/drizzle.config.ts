//

import type { Config } from "drizzle-kit";
import { env } from "node:process";

//

export default {
  dbCredentials: {
    url:
      env["DATABASE_URL"] ||
      "postgresql://postgres:postgres@localhost:6543/postgres?schema=public",
  },
  dialect: "postgresql",
  introspect: { casing: "preserve" },
  out: "src/drizzle",
  schema: "src/drizzle/schema.ts",
  strict: true,
  verbose: true,
} satisfies Config;
