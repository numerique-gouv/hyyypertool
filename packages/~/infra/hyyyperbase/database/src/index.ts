export { drizzle } from "drizzle-orm/node-postgres";
