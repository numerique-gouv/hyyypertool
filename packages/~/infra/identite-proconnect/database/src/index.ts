//

import type { PgDatabase, PgQueryResultHKT } from "drizzle-orm/pg-core";
import Pg from "pg";
import * as schema from "./drizzle.schema";

//

export { drizzle } from "drizzle-orm/node-postgres";
export * from "./types";
export { schema };

export type IdentiteProconnect_PgDatabase = PgDatabase<
  PgQueryResultHKT,
  typeof schema
>;
export type IdentiteProconnectSdkDatabaseCradle = {
  pg_client: Pg.Pool;
};

export type IdentiteProconnectDatabaseCradle = {
  pg: IdentiteProconnect_PgDatabase;
};
export const Pool = Pg.Pool;
