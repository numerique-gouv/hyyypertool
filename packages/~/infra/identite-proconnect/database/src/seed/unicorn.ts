//

import type { MCP_EmailDomain_Type } from "@~/identite-proconnect.lib/identite-proconnect.d";
import { eq } from "drizzle-orm";
import { schema, type IdentiteProconnect_PgDatabase } from "..";

//

// Uses https://magicalunicornlife.com/the-best-unicorn-names/

//
export async function create_unicorn_organization(
  pg: IdentiteProconnect_PgDatabase,
) {
  const [{ id: organization_id }] = await pg
    .insert(schema.organizations)
    .values({
      cached_libelle: "ðŸ¦„ libelle",
      siret: "ðŸ¦„ siret",
    })
    .returning({ id: schema.organizations.id });

  await pg.insert(schema.email_domains).values({
    domain: "unicorn.xyz",
    organization_id,
    verification_type: "verified" as MCP_EmailDomain_Type,
  });

  return organization_id;
}

//

export async function create_adora_pony_user(
  pg: IdentiteProconnect_PgDatabase,
) {
  const [{ id: user_id }] = await pg
    .insert(schema.users)
    .values({
      created_at: new Date().toISOString(),
      email: "adora.pony@unicorn.xyz",
      family_name: "Pony",
      given_name: "Adora",
      updated_at: new Date().toISOString(),
    })
    .returning({ id: schema.users.id });

  return user_id;
}

export async function create_adora_pony_moderation(
  pg: IdentiteProconnect_PgDatabase,
  moderation: Omit<
    typeof schema.moderations.$inferInsert,
    "organization_id" | "user_id"
  >,
) {
  const unicorn_organization = await pg.query.organizations.findFirst({
    columns: { id: true },
    where: eq(schema.organizations.siret, "ðŸ¦„ siret"),
  });
  const adora_pony_user = await pg.query.users.findFirst({
    columns: { id: true },
    where: eq(schema.users.email, "adora.pony@unicorn.xyz"),
  });
  const [{ id: moderation_id }] = await pg
    .insert(schema.moderations)
    .values({
      ...moderation,
      organization_id: unicorn_organization!.id,
      user_id: adora_pony_user!.id,
    })
    .returning({ id: schema.moderations.id });
  return moderation_id;
}

//

export async function create_pink_diamond_user(
  pg: IdentiteProconnect_PgDatabase,
) {
  const [{ id: user_id }] = await pg
    .insert(schema.users)
    .values({
      created_at: new Date().toISOString(),
      email: "pink.diamond@unicorn.xyz",
      family_name: "Diamond",
      given_name: "Pink",
      updated_at: new Date().toISOString(),
    })
    .returning({ id: schema.users.id });

  return user_id;
}

export async function create_red_diamond_user(
  pg: IdentiteProconnect_PgDatabase,
) {
  const [{ id: user_id }] = await pg
    .insert(schema.users)
    .values({
      created_at: new Date().toISOString(),
      email: "red.diamond@unicorn.xyz",
      family_name: "Diamond",
      given_name: "Red",
      updated_at: new Date().toISOString(),
    })
    .returning({ id: schema.users.id });

  return user_id;
}
