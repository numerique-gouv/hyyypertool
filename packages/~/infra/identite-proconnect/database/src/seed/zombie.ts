//

import { schema, type IdentiteProconnect_PgDatabase } from "..";

//

export async function create_zombie_organization(
  pg: IdentiteProconnect_PgDatabase,
) {
  const [{ id: organization_id }] = await pg
    .insert(schema.organizations)
    .values({
      cached_libelle: "üßü‚Äç‚ôÇÔ∏è libelle",
      siret: "üßü‚Äç‚ôÇÔ∏è siret",
      cached_est_active: false,
    })
    .returning({ id: schema.organizations.id });

  await pg.insert(schema.email_domains).values({
    domain: "zombie.corn",
    organization_id,
    verification_type: null,
  });

  return organization_id;
}
