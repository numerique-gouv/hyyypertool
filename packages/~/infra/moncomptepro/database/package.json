{
  "name": "@~/moncomptepro.database",
  "version": "1.0.0",
  "private": true,
  "type": "module",
  "exports": {
    ".": {
      "default": "./src/index.ts"
    },
    "./testing": {
      "default": "./src/testing/index.ts"
    },
    "./*": {
      "default": "./src/*.ts"
    }
  },
  "scripts": {
    "introspect": "bun x drizzle introspect:pg",
    "migration": "bun run bin/migration.ts",
    "seed": "bun run bin/seed.ts",
    "studio": "drizzle-kit studio"
  },
  "dependencies": {
    "@~/app.core": "workspace:*",
    "@~/moncomptepro.lib": "workspace:*",
    "consola": "3.2.3",
    "drizzle-orm": "0.36.4",
    "hono": "4.6.12",
    "pg": "8.13.0"
  },
  "devDependencies": {
    "@electric-sql/pglite": "0.2.14",
    "@types/pg": "8.11.10",
    "@~/config.typescript": "workspace:*",
    "drizzle-kit": "0.28.1"
  }
}
