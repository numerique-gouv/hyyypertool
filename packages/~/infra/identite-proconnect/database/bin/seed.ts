//

import { seed } from "../src/seed";

//

await seed();
