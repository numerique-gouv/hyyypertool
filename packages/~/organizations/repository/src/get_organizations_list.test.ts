//

import { create_cactus_organization } from "@~/moncomptepro.database/seed/captus";
import { create_unicorn_organization } from "@~/moncomptepro.database/seed/unicorn";
import { create_zombie_organization } from "@~/moncomptepro.database/seed/zombie";
import { empty_database, migrate, pg } from "@~/moncomptepro.database/testing";
import { beforeAll, beforeEach, expect, test } from "bun:test";
import { get_organizations_list } from "./get_organizations_list";

//

beforeAll(migrate);
beforeEach(empty_database);

//

test("returns no organizations", async () => {
  const result = await get_organizations_list(pg, {});
  expect(result).toMatchSnapshot();
});

test("returns 1 organization", async () => {
  await create_unicorn_organization(pg);
  const result = await get_organizations_list(pg, {});
  expect(result).toMatchSnapshot();
});

test("returns 3 organizations", async () => {
  await create_cactus_organization(pg);
  await create_unicorn_organization(pg);
  await create_zombie_organization(pg);

  const result = await get_organizations_list(pg, {});

  expect(result).toMatchSnapshot();
});

test("search ðŸ¦„ libelle", async () => {
  await create_cactus_organization(pg);
  await create_unicorn_organization(pg);
  await create_zombie_organization(pg);

  const result = await get_organizations_list(pg, {
    search: "ðŸ¦„ libelle",
  });

  expect(result).toMatchSnapshot();
});

test("search ðŸ¦„ siret", async () => {
  await create_cactus_organization(pg);
  await create_unicorn_organization(pg);
  await create_zombie_organization(pg);

  const result = await get_organizations_list(pg, {
    search: "ðŸ¦„ siret",
  });

  expect(result).toMatchSnapshot();
});
