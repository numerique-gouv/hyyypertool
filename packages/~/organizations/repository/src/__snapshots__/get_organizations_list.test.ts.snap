// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`returns no organizations 1`] = `
{
  "count": 0,
  "organizations": [],
}
`;

exports[`returns 1 organization 1`] = `
{
  "count": 1,
  "organizations": [
    {
      "cached_code_officiel_geographique": null,
      "cached_libelle": "🦄 libelle",
      "created_at": "1970-01-01 00:00:00+00",
      "email_domains": [
        {
          "domain": "unicorn.xyz",
        },
      ],
      "id": 1,
      "siret": "🦄 siret",
    },
  ],
}
`;

exports[`returns 3 organizations 1`] = `
{
  "count": 3,
  "organizations": [
    {
      "cached_code_officiel_geographique": null,
      "cached_libelle": "🌵 libelle",
      "created_at": "1970-01-01 00:00:00+00",
      "email_domains": [
        {
          "domain": "cactus.corn",
        },
      ],
      "id": 1,
      "siret": "🌵 siret",
    },
    {
      "cached_code_officiel_geographique": null,
      "cached_libelle": "🦄 libelle",
      "created_at": "1970-01-01 00:00:00+00",
      "email_domains": [
        {
          "domain": "unicorn.xyz",
        },
      ],
      "id": 2,
      "siret": "🦄 siret",
    },
    {
      "cached_code_officiel_geographique": null,
      "cached_libelle": "🧟‍♂️ libelle",
      "created_at": "1970-01-01 00:00:00+00",
      "email_domains": [
        {
          "domain": "zombie.corn",
        },
      ],
      "id": 3,
      "siret": "🧟‍♂️ siret",
    },
  ],
}
`;

exports[`search 🦄 libelle 1`] = `
{
  "count": 1,
  "organizations": [
    {
      "cached_code_officiel_geographique": null,
      "cached_libelle": "🦄 libelle",
      "created_at": "1970-01-01 00:00:00+00",
      "email_domains": [
        {
          "domain": "unicorn.xyz",
        },
      ],
      "id": 2,
      "siret": "🦄 siret",
    },
  ],
}
`;

exports[`search 🦄 siret 1`] = `
{
  "count": 1,
  "organizations": [
    {
      "cached_code_officiel_geographique": null,
      "cached_libelle": "🦄 libelle",
      "created_at": "1970-01-01 00:00:00+00",
      "email_domains": [
        {
          "domain": "unicorn.xyz",
        },
      ],
      "id": 2,
      "siret": "🦄 siret",
    },
  ],
}
`;
